graph-partitioning/README.rst

```diff
@@ -0,0 +1,85 @@
+
+.. image:: readme_imgs/constraint_1.png
+
+For a QUBO, we need our constraints to be represented by mathematical expressions that are satisfied at their minimum value.  For this constraint, we can use the following expression that has a minimum value of 0 that occurs when Subset 1 has size exactly `|V|`/2.
+
+.. image:: readme_imgs/constraint_2.png
+
+To simplify this expression and determine the coefficients for our QUBO equation, we first multiply out the expression.
+
+.. image:: readme_imgs/constraint_3.png
+
+Next we can simplify this expression down to linear and quadratic terms for our QUBO.  Recall that for binary variables we can replace any squared term with a linear term (since 0^2=0 and 1^2=1), and we can remove any constant terms in our QUBO.  This results in the following final expression for our constraint.
+
+.. image:: readme_imgs/constraint_4.png
+
+To combine our objective and constraints into a single QUBO expression, we simply add together the objective function and our constraint (multiplied by gamma, the Lagrange parameter).  
+
+.. image:: readme_imgs/final_QUBO.png
+   :align: center
+   :height: 100
+
+In the code, we create the Q matrix for this QUBO as a dictionary iteratively, looping over the edges and nodes in our graph just as we see in the summation of our QUBO expression.
+
+This demo generates an Erdos-Renyi random graph using the ``networkx`` package for our problem instance [1]_. There are three parameters to be set by the user in this code:  chain strength, number of reads, and gamma.  Since this is a relatively large problem, we set a large number of reads (``num_reads = 1000``).  
+
+For chain strength, we examine the entries in our Q matrix and choose a relatively large number to enforce chains in our embedding.  For this problem, our largest matrix entries are the linear coefficients (approximately size ``gamma*(1-1*len(G.nodes))``) and so a value of ``gamma*len(G.nodes)`` is chosen on line 51 for ``chain_strength``.
+
+For gamma, our Lagrange parameter, a good number to start with is an estimate for your optimal objective function value.  Since we are utilizing Erdos-Renyi random graphs, we assume that our graph will have about 20% of the number of cut edges that a complete graph would have (since our edge probability in G is also 20%).  Since a complete graph on 40 nodes has 720 edges and any partition into equal subsets results in a cut of size 400.  Since we are examining an Erdos-Renyi random graph with an edge probability of 20%, we can estimate a cut of size approximately 20% of 400, or 80 edges. This leads us to try a Lagrange parameter of 80, which is implemented in our program with ``gamma=80``.  
```

Comment:
Typo? A complete graph with 40 nodes should have 780 edges rather than 720?

Just a suggestion: Maybe additional explanation to cut of size 400. Perhaps `.. cut of size 400 (i.e. all 20 nodes in Subset0 has edges with all 20 nodes in Subset1, hence ``20 * 20 = 400``) `